src/nvim/screen.c

```diff
@@ -6079,8 +6079,10 @@ void grid_alloc(ScreenGrid *grid, int rows, int columns, bool copy, bool valid)
 {
   int new_row;
   ScreenGrid new = *grid;
-
-  size_t ncells = (size_t)((rows+1) * columns);
+  /* Since amount of cells could not be negative we expect both
+   * rows and columns to be positive */
+  assert(rows >= 0 && columns >= 0);
+  size_t ncells = ((size_t)rows + 1) * columns;
```

Comment:
the `+ 1` can be removed. it was used for `current_screenline`, which now is dedicated `linebuf_char` buffer.

Reply:
done

Reply:
> the `+ 1` can be removed. it was used for `current_screenline`, which now is dedicated `linebuf_char` buffer.

that explanation should be pasted into the commit message.